### ISSUE TITLE
Google::Cloud::Logging::Entry overrides ruby reserved word

### ISSUE THREAD

**Issue by matti** [GoogleCloudPlatform/google-cloud-ruby#1798]
https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-logging/lib/google/cloud/logging/entry/http_request.rb#L36

vs

```
$ irb
2.4.1 :001 > class Kakka
2.4.1 :002?>   def kekke
2.4.1 :003?>     end
2.4.1 :004?>   end
 => :kekke
2.4.1 :005 > k = Kakka.new
 => #<Kakka:0x007f84ae08acc0>
2.4.1 :006 > k.method
ArgumentError: wrong number of arguments (given 0, expected 1)
	from (irb):6:in `method'
	from (irb):6
	from /Users/mpa/.rvm/rubies/ruby-2.4.1/bin/irb:11:in `<main>'
2.4.1 :007 > k.method :kekke
 => #<Method: Kakka#kekke>
```

**Comment by blowmage** [comment]
I think this issue's title is misleading, as `method` is not a reserved word in ruby. I believe it is only appropriate to say that is `Google::Cloud::Logging::Entry::HttpRequest#method` overrides `Object#method`. Is this a problem?

**Comment by matti** [comment]
Yes - because it Google::Cloud::Logging::Entry::HttpRequest#method gets called with an argument here https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-debugger/lib/google/cloud/debugger/breakpoint/evaluator.rb#L940

and then this is rescued without checking the exception class in the next line: https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-debugger/lib/google/cloud/debugger/breakpoint/evaluator.rb#L941

**Comment by quartzmo** [comment]
@dazuma Do you have any input/opinion on this issue?

**Comment by dazuma** [comment]
That code in the debugger should be checking the exception type rather than rescuing the world. That should be fixed over in the debugger. I'll open a separate issue for that.

However, that said, overriding methods of `Object` is probably not a good idea regardless. I'd advocate for renaming the accessor to `request_method` since that's the name of the underlying proto field anyway. We can take advantage of the arity difference to maintain backward compatibility: if an argument is given, pass through to super, otherwise alias to the new name.

**Comment by quartzmo** [comment]
> I'd advocate for renaming the accessor to request_method since that's the name of the underlying proto field anyway. We can take advantage of the arity difference to maintain backward compatibility: if an argument is given, pass through to super, otherwise alias to the new name.

@dazuma SGTM. Please self-assign if this is something you want to do, otherwise I will take a shot at it.

**Comment by blowmage** [comment]
This is released in [Logging 1.3.0](https://github.com/GoogleCloudPlatform/google-cloud-ruby/releases/tag/google-cloud-logging%2Fv1.3.0). You can continue to use `HttpRequest#method` to  get the logging value, but you can now get a `Method` object when passing an argument. This maintains backwards compatibility while fixing the issues described in this issue.

Even though you can continue to use `HttpRequest#method` in this manner, we have marked it deprecated and encourage using `HttpRequest#request_method` instead. Thank you for your contribution!
